### PR TITLE
Added DNF configuration and Multimedia Support for fedora and other Redhat Distros

### PR DESCRIPTION
**Pull request by Angxddeep**
# Pull Request

## DNF Configuration & Multimedia Codecs
Just some essential DNF changes to at least make it usable and faster by a margin.
Also added installation support for multimedia codecs

## Type of Change
-  New feature

## Testing
Checked on my System, it works

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

File: src/commands/system-setup/fedora/configure-dnf.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/bin/sh -e
+
+. "$(dirname "$0")/../../common-script.sh"
+
+configureDNF() {
+    case $PACKAGER in
+        dnf)
+            $ESCALATION_TOOL sed -i '/^max_parallel_downloads=/c\max_parallel_downloads=10' /etc/dnf/dnf.conf || echo 'max_parallel_downloads=10' >> /etc/dnf/dnf.conf
```
Comment (nnyyxxxx) [inline src/commands/system-setup/fedora/configure-dnf.sh:8]: I'd recommend turning this down to 5, not everyone has a good internet connection, and as such putting it up to 10 can cause more harm than good.

File: src/commands/system-setup/fedora/configure-dnf.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+            echo "fastestmirror=True" | $ESCALATION_TOOL tee -a /etc/dnf/dnf.conf > /dev/null
+            echo "defaultyes=True" | $ESCALATION_TOOL tee -a /etc/dnf/dnf.conf > /dev/null
+            
+            $ESCALATION_TOOL "$PACKAGER" -y install dnf-plugins-core
+            ;;
+        *)
```
Comment (nnyyxxxx) [inline src/commands/system-setup/fedora/configure-dnf.sh:14]: Consider adding an error message for unsupported package managers.

File: src/commands/system-setup/fedora/configure-dnf.sh (hunk 1 above)
Comment (Angxddeep) [reply src/commands/system-setup/fedora/configure-dnf.sh:8]: I have set it to 20, and I just have a 100mbps connection the thing with fedora is that most of the updates are just some python libraries of 200-300Kb. I don't know what the deal is with Fedora servers, but I can only get maximum of 400 KBps from a single package here in India it's because of the parallel downloading which makes it to 10-15 MBps at least

Comment (nnyyxxxx) [reply src/commands/system-setup/fedora/configure-dnf.sh:8]: Either way I still think 10 is too high for the majority.

Comment (Angxddeep) [reply src/commands/system-setup/fedora/configure-dnf.sh:8]: SO here the deal, let's make it 8 it's the middle ground and my birthdate

Comment (nnyyxxxx) [reply src/commands/system-setup/fedora/configure-dnf.sh:8]: @Angxddeep 8 Isn't much of a change, and I still think that is way too high; 5 is the best option here...

File: src/commands/system-setup/fedora/multimedia-codecs.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
+#!/bin/sh -e
+
+. "$(dirname "$0")/../../common-script.sh"
+
+multimedia() {
+    case $PACKAGER in
+        dnf)
+            if [[ -e /etc/yum.repos.d/rpmfusion-free.repo && -e /etc/yum.repos.d/rpmfusion-nonfree.repo ]]; then
```
Comment (nnyyxxxx) [inline src/commands/system-setup/fedora/multimedia-codecs.sh:8]: This is not posix compliant

File: src/commands/system-setup/fedora/multimedia-codecs.sh [hunk 4]
```diff
@@ -0,0 +1,23 @@
+#!/bin/sh -e
+
+. "$(dirname "$0")/../../common-script.sh"
```
Comment (nnyyxxxx) [inline src/commands/system-setup/fedora/multimedia-codecs.sh:3]: This is not posix compliant.

File: src/commands/system-setup/fedora/configure-dnf.sh [hunk 5]
```diff
@@ -0,0 +1,21 @@
+#!/bin/sh -e
+
+. "$(dirname "$0")/../../common-script.sh"
```
Comment (nnyyxxxx) [inline src/commands/system-setup/fedora/configure-dnf.sh:3]: This is not posix compliant.

**Comment by ChrisTitusTech** [comment]
Sorry for the inconvenience. We had a massive restructure of the codebase to improve future development. Because of this can you update your PR to the new structure. Thank you for your assistance and contribution.

**Comment by Angxddeep** [comment]
> Sorry for the inconvenience. We had a massive restructure of the codebase to improve future development. Because of this can you update your PR to the new structure. Thank you for your assistance and contribution.

DONE✅

File: tabs/system-setup/fedora/rpm-fusion-setup.sh [hunk 6]
```diff
@@ -1,6 +1,6 @@
 #!/bin/sh -e
 
-. "$(dirname "$0")/../../common-script.sh"
```
Comment (nnyyxxxx) [inline tabs/system-setup/fedora/rpm-fusion-setup.sh:3]: You should undo this change, It's going to conflict with #282 & It's a duplicate. The change is already proposed In #282
